src/libs/actions/Policy.ts

```diff
@@ -2178,6 +2194,42 @@ function createWorkspaceFromIOUPayment(iouReport: Report | EmptyObject): string
     return policyID;
 }
 
+function setWorkspaceCategoryEnabled(policyID: string, categoriesToUpdate: Record<string, {name: string; enabled: boolean}>) {
+    const policyCategories = allCategoryPolicies?.[policyID] ?? {};
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: {
+                    ...Object.keys(categoriesToUpdate).reduce<Record<string, PolicyCategory>>((acc, key) => {
+                        acc[key] = {...policyCategories[key], ...categoriesToUpdate[key]};
+
+                        return acc;
+                    }, {}),
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: {
+                    ...policyCategories,
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        categories: Object.keys(categoriesToUpdate).map((key) => categoriesToUpdate[key]),
```

Comment:
I think this solves the json issues you were seeing on this request:
```suggestion
        categories: JSON.stringify(Object.keys(categoriesToUpdate).map((key) => categoriesToUpdate[key])),
```

Reply:
Cool, too bad we lost the type of this parameter because it will now be a string type